flake.nix

```diff
@@ -19,89 +17,309 @@
+              help = "Rebuild host target on save";
+              category = "watch";
+              command = "${cargo} watch --clear";
+            }
+            {
+              name = "watch:build:wasm";
+              help = "Rebuild host target on save";
+              category = "watch";
+              command = "${cargo} watch --clear --features=serde -- cargo build --target=wasm32-unknown-unknown";
+            }
+            {
+              name = "watch:lint";
+              help = "Lint on save";
+              category = "watch";
+              command = "${cargo} watch --clear --exec clippy";
+            }
+            {
+              name = "watch:lint:pedantic";
+              help = "Pedantic lint on save";
+              category = "watch";
+              command = "${cargo} watch --clear --exec 'clippy -- -W clippy::pedantic'";
+            }
+            {
+              name = "watch:test:host";
+              help = "Run all tests on save";
+              category = "watch";
+              command = "${cargo} watch --clear --exec test";
+            }
+            {
+              name = "watch:test:docs:host";
+              help = "Run all tests on save";
+              category = "watch";
+              command = "${cargo} watch --clear --exec test";
+            }
+            # Test
+            {
+              name = "test:all";
+              help = "Run Cargo tests";
+              category = "test";
+              command = "test:host && test:docs && test:wasm";
+            }
+            {
+              name = "test:host";
+              help = "Run Cargo tests for host target";
+              category = "test";
+              command = "${cargo} test";
+            }
+            {
+              name = "test:wasm";
+              help = "Run wasm-pack tests on all targets";
+              category = "test";
+              command = "test:wasm:node && test:wasm:chrome";
+            }
+            {
+              name = "test:wasm:nodejs";
+              help = "Run wasm-pack tests in Node.js";
+              category = "test";
+              command = "${wasm-pack} test --node";
+            }
+            {
+              name = "test:wasm:chrome";
+              help = "Run wasm-pack tests in headless Chrome";
+              category = "test";
+              command = "${wasm-pack} test --headless --chrome";
+            }
+            {
+              name = "test:docs";
+              help = "Run Cargo doctests";
+              category = "test";
+              command = "${cargo} test --doc";
+            }
+            # Docs
+            {
+              name = "docs";
+              help = "[DEFAULT]: Open refreshed docs";
+              category = "dev";
+              command = "docs:open";
+            }
+            {
+              name = "docs:build";
+              help = "Refresh the docs";
+              category = "dev";
+              command = "${cargo} doc --features=mermaid_docs";
```

Comment:
Annoying workaround, but AFAICT the `pkgs.cargo` here and `cargo` in the shell are different `cargo`s... my guess is that the shell one probably one comes from the toolchain, but I haven't debugged it much yet